storage/flux/table.gen.go

```diff
@@ -97,6 +97,65 @@ func (t *floatTable) advance() bool {
 	return true
 }
 
+// window table
+type floatWindowTable struct {
+	floatTable
+	windowEvery int64
+}
+
+func newFloatWindowTable(
+	done chan struct{},
+	cur cursors.FloatArrayCursor,
+	bounds execute.Bounds,
+	every int64,
+	key flux.GroupKey,
+	cols []flux.ColMeta,
+	tags models.Tags,
+	defs [][]byte,
+	cache *tagsCache,
+	alloc *memory.Allocator,
+) *floatWindowTable {
+	t := &floatWindowTable{
+		floatTable: floatTable{
+			table: newTable(done, bounds, key, cols, defs, cache, alloc),
+			cur:   cur,
+		},
+		windowEvery: every,
+	}
+	t.readTags(tags)
+	t.advance()
+
+	return t
+}
+
+func (t *floatWindowTable) advance() bool {
+	a := t.cur.Next()
+	l := a.Len()
+	if l == 0 {
+		return false
+	}
+
+	// Retrieve the buffer for the data to avoid allocating
+	// additional slices. If the buffer is still being used
+	// because the references were retained, then we will
+	// allocate a new buffer.
+	cr := t.allocateBuffer(l)
+	// regain the window start time from the window end time
+	rangeStart := int64(t.bounds.Start)
+	startTimes := make([]int64, len(a.Timestamps))
```

Comment:
This isn't the optimal way to do this and it will result in another untracked allocation. We should create the start column using the arrow builders directly.

```go
startB := arrow.NewIntBuilder(t.alloc)
startB.Resize(len(a.Timestamps))
for i := 0; i < a.Len(); i++ {
    ts := a.Timestamps[i] - t.windowEvery
    if ts < int64(t.bounds.Start) {
        ts = int64(t.bounds.Start)
    }
    startB.Append(ts)
}
cr.cols[startColIdx] = startB.NewArray()
```

The reason we don't do the same for the stop or value columns is mostly because we get those arrays from protobuf so we don't control their allocations. Because the start column is something we create that doesn't exist on the wire, it will minimize copies and track allocations better if we use the arrow builder.

Reply:
That is a good point!

Reply:
> The current method will result in an untracked allocation.

This seems to a be a go-internals sort of knowledge point. Could you teach me to fish here? How did you identify that untracked allocation? What am I missing?

Reply:
It's the `make()` call which is standard for Go, but our memory tracking system won't recognize it so the allocator won't prevent the allocation. When you use the arrow builders, it uses `make()` under the hood, but it track the amount of memory it has used to construct arrays. It's a very flux-specific thing that isn't part of Go itself.